externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/spark/kyuubi/SparkSQLEngineListener.scala

```diff
@@ -45,21 +46,29 @@ class SparkSQLEngineListener(server: Serverable) extends SparkListener with Logg
   override def onJobEnd(jobEnd: SparkListenerJobEnd): Unit = {
    jobEnd.jobResult match {
      case JobFailed(e) if e != null =>
+       // get valuable exception
+       val ve = e match {
+         case se: SparkException if se.getCause != null =>
+           se.getCause
+         case _ => e
+       }
+
        var deregisterInfo: Option[String] = None
-       if (deregisterExceptions.exists(_.equals(e.getClass.getCanonicalName))) {
```

Comment:
On second thought, we can merge message and stack trace together with `KyuubiSQLException.stringlifyException` and check once? we can save one config for better user experience.

WDYT @pan3793

Reply:
I'm ok with it, but how to name this conf?

Reply:
we can keep the `message` as is, update the config doc and drop another

Reply:
Can you make a separate PR for this? @turboFei

Reply:
sure